test/managers/hub/unit/SimplePriceManager.t.sol

```diff
@@ -379,4 +388,96 @@ contract SimplePriceManagerOnTransferTest is SimplePriceManagerTest {
         vm.prank(caller);
         priceManager.onTransfer(POOL_A, SC_2, CENTRIFUGE_ID_1, CENTRIFUGE_ID_2, 50);
     }
+
+    function testOnTransferWithUpdate() public {
+        uint128 sharesTransferred = 50;
+
+        (, uint128 initialGlobalIssuance) = priceManager.metrics(POOL_A);
+        assertEq(initialGlobalIssuance, 300); // 100 + 200
+
+        vm.mockCall(
+            shareClassManager,
+            abi.encodeWithSelector(IShareClassManager.issuance.selector, POOL_A, SC_1, CENTRIFUGE_ID_1),
+            abi.encode(50) // 100 - 50
+        );
+        vm.mockCall(
+            shareClassManager,
+            abi.encodeWithSelector(IShareClassManager.issuance.selector, POOL_A, SC_1, CENTRIFUGE_ID_2),
+            abi.encode(250) // 200 + 50
+        );
+
+        vm.prank(caller);
+        priceManager.onTransfer(POOL_A, SC_1, CENTRIFUGE_ID_1, CENTRIFUGE_ID_2, sharesTransferred);
+
+        vm.prank(caller);
+        priceManager.onUpdate(POOL_A, SC_1, CENTRIFUGE_ID_1, 1000);
+
+        (, uint128 globalIssuance) = priceManager.metrics(POOL_A);
+        (, uint128 fromIssuance, uint128 fromTransferredIn, uint128 fromTransferredOut,,) =
+            priceManager.networkMetrics(POOL_A, CENTRIFUGE_ID_1);
+        (, uint128 toIssuance, uint128 toTransferredIn, uint128 toTransferredOut,,) =
+            priceManager.networkMetrics(POOL_A, CENTRIFUGE_ID_2);
+
+        // fromIssuance is now the same as in ShareClassManager
+        // global issuance remains unchanged
+        assertEq(globalIssuance, 300);
+
+        // toIssuance is still stale until we call onUpdate for that network
+        assertEq(fromIssuance, 50);
+        assertEq(toIssuance, 200);
+
+        assertEq(fromTransferredOut, 0);
+        assertEq(fromTransferredIn, 0);
+        assertEq(toTransferredIn, 50);
+        assertEq(toTransferredOut, 0);
+
+        // Update the first network again, just to make sure it doesn't break anything
+        vm.prank(caller);
+        priceManager.onUpdate(POOL_A, SC_1, CENTRIFUGE_ID_1, 1000);
+
+        (, globalIssuance) = priceManager.metrics(POOL_A);
+        (, fromIssuance, fromTransferredIn, fromTransferredOut,,) = priceManager.networkMetrics(POOL_A, CENTRIFUGE_ID_1);
+        (, toIssuance, toTransferredIn, toTransferredOut,,) = priceManager.networkMetrics(POOL_A, CENTRIFUGE_ID_2);
+
+        assertEq(globalIssuance, 300);
+        assertEq(fromIssuance, 50);
+        assertEq(toIssuance, 200);
+        assertEq(fromTransferredOut, 0);
+        assertEq(fromTransferredIn, 0);
+        assertEq(toTransferredIn, 50);
```

Comment:
Q: IIUC, this is not 0 because we haven't called onUpdate for network 2 yet right?

Reply:
yes exactly